Tests.Unit.Core.ApplicationServices/ApplicationServices/ItInterfaceServiceTest.cs

```diff
@@ -270,6 +283,144 @@ public void Delete_Returns_Ok_And_Raises_Domain_Event()
+            var orgId = A<int>();
+            var rightsHolderOrgs = new List<int>() { orgId };
+            var itInterface = new ItInterface()
+            {
+                Uuid = interfaceUuid
+            };
+
+            _authorizationContext.Setup(x => x.GetCrossOrganizationReadAccess()).Returns(CrossOrganizationDataReadAccessLevel.RightsHolder);
+            _userContext.Setup(x => x.GetOrganizationIdsWhereHasRole(OrganizationRole.RightsHolderAccess)).Returns(rightsHolderOrgs);
+            _repository.Setup(x => x.GetInterfacesFromRightsHolderOrganizations(rightsHolderOrgs)).Returns(new List<ItInterface>() { itInterface }.AsQueryable());
+
+            //Act
+            var result = _sut.GetAvailableInterfaces();
+
+            //Assert
+            var interfaceResult = Assert.Single(result.ToList());
+            Assert.Equal(interfaceUuid, interfaceResult.Uuid);
+        }
+
+        [Fact]
+        public void GetAvailableInterfaces_Returns_RightsholderInterfaces_From_Specific_Rightsholder_If_User_Has_Rightsholderaccess()
+        {
+            //Arrange
+            var org1 = new Organization()
+            {
+                Id = A<int>(),
+                Uuid = Guid.NewGuid()
+            };
+            var org2 = new Organization()
+            {
+                Id = A<int>(),
+                Uuid = Guid.NewGuid()
+            };
+            var rightsHolderOrgs = new List<int>() {
+                org1.Id,
+                org2.Id
+            };
+            var itInterface1 = new ItInterface()
+            {
+                Uuid = Guid.NewGuid(),
+                ExhibitedBy = new ItInterfaceExhibit()
+                {
+                    ItSystemId = A<int>(),
+                    ItSystem = new ItSystem()
+                    {
+                        BelongsTo = org1,
+                        BelongsToId = org1.Id
+                    }
+                }
+            };
+            var itInterface2 = new ItInterface()
+            {
+                Uuid = Guid.NewGuid(),
+                ExhibitedBy = new ItInterfaceExhibit()
+                {
+                    ItSystemId = A<int>(),
+                    ItSystem = new ItSystem()
+                    {
+                        BelongsTo = org2,
+                        BelongsToId = org2.Id
+                    }
+                }
+            };
+
+            _authorizationContext.Setup(x => x.GetCrossOrganizationReadAccess()).Returns(CrossOrganizationDataReadAccessLevel.RightsHolder);
+            _userContext.Setup(x => x.GetOrganizationIdsWhereHasRole(OrganizationRole.RightsHolderAccess)).Returns(rightsHolderOrgs);
+            _repository.Setup(x => x.GetInterfacesFromRightsHolderOrganizations(rightsHolderOrgs)).Returns(new List<ItInterface>() { itInterface1, itInterface2 }.AsQueryable());
+
+            var refinements = new List<IDomainQuery<ItInterface>>();
+
+            refinements.Add(new QueryByRightsHolder(org1.Uuid));
```

Comment:
du behøves faktisk ikke at være så konkret i din query. Du kan lave mock af IDomainQuery i stedet og så lave setup for den hvor den f.eks. bare tager 1 element fra input

Så tester du kun at input rent faktisk bruges og ikke en konkret implementation